internal/core/src/segcore/IndexConfigGenerator.cpp

```diff
@@ -106,8 +111,24 @@ VecIndexConfig::GetMetricType() noexcept {
 }
 
 knowhere::Json
-VecIndexConfig::GetBuildBaseParams() {
-    return build_params_;
+VecIndexConfig::GetBuildBaseParams(DataType data_type) {
+    if (data_type == DataType::VECTOR_BFLOAT16 &&
+        build_params_[knowhere::indexparam::REFINE_TYPE] ==
+            knowhere::RefineType::FLOAT16_QUANT) {
+        auto out_config = build_params_;
+        out_config[knowhere::indexparam::REFINE_TYPE] =
+            knowhere::RefineType::BFLOAT16_QUANT;
+        return out_config;
+    } else if (data_type == DataType::VECTOR_FLOAT16 &&
+               build_params_[knowhere::indexparam::REFINE_TYPE] ==
+                   knowhere::RefineType::BFLOAT16_QUANT) {
+        auto out_config = build_params_;
+        out_config[knowhere::indexparam::REFINE_TYPE] =
```

Comment:
Is this dealing with a scenario where the parameters are not expected? BF16 but with FP16?